kotest-assertions/src/jvmMain/kotlin/io/kotest/matchers/date/timestamp.kt

```diff
@@ -0,0 +1,74 @@
+package io.kotest.matchers.date
+
+import io.kotest.Matcher
+import io.kotest.MatcherResult
+import io.kotest.should
+import io.kotest.shouldNot
+import java.sql.Timestamp
+
+fun sameTimestamp(timestamp: Timestamp) = object: Matcher<Timestamp> {
+   override fun test(value: Timestamp): MatcherResult {
+      return MatcherResult(
+         timestamp.equals(value),
+         {"Expected $timestamp to be equal $value, but it's not."},
+         {"$timestamp not expected to be equal $value, but it is equal"}
+      )
+   }
+}
+
+fun afterTimestamp(timestamp: Timestamp) = object: Matcher<Timestamp>  {
+   override fun test(value: Timestamp): MatcherResult {
+      return MatcherResult(
+         value.after(timestamp),
+         {"Expected $value to be after $timestamp, but it's not."},
+         {"$value is not expected to be after $timestamp."}
+      )
+   }
+}
+
+fun beforeTimestamp(timestamp: Timestamp) = object: Matcher<Timestamp>  {
+   override fun test(value: Timestamp): MatcherResult {
+      return MatcherResult(
+         value.before(timestamp),
+         {"Expected $value to be before $timestamp, but it's not."},
+         {"$value is not expected to be before $timestamp."}
+      )
+   }
+}
+
+
+/**
+ * Assert that [Timestamp] is equal to [anotherTimestamp].
+ * @see [shouldNotBe]
+ * */
+infix fun Timestamp.shouldBe(anotherTimestamp: Timestamp) = this should sameTimestamp(anotherTimestamp)
```

Comment:
Ok, on a further review, I believe we don't want a `shouldBe` and a `shouldNotBe` for this guy.

We already have the default `shouldBe` matcher that compares using `==`, so this is redundant and unnecessary.

Could you please take them out of this commit and from the documentation commit?

Reply:
One thing here is error message which is specific to timestamp when use Timestamp.shouldBe while in the regular one error message is generic.

Reply:
The default should be message is `expected: $expected but was: $actual`
I don't think the message difference has enough relevance to justify this change

What do you think?

Reply:
Yes now i also think we don't need shouldBe and shouldNotBe i will remove those methods.

Reply:
Done